src/main/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWork.java

```diff
@@ -55,22 +60,39 @@ protected void doRun() {
 
     private void cleanCloud(ComputeEngineCloud cloud) {
         logger.log(Level.FINEST, "Cleaning cloud " + cloud.getCloudName());
-        List<Instance> remoteInstances = findRemoteInstances(cloud);
+        ComputeClientV2 clientV2 = ComputeClientV2.createFromComputeEngineCloud(cloud);
+        List<Instance> remoteInstances = findRunningRemoteInstances(clientV2);
         Set<String> localInstances = findLocalInstances(cloud);
+        if (!(localInstances.isEmpty() || remoteInstances.isEmpty())) {
+            updateLocalInstancesLabel(clientV2, localInstances, remoteInstances);
+        }
         remoteInstances.stream()
                 .filter(remote -> isOrphaned(remote, localInstances))
                 .forEach(remote -> terminateInstance(remote, cloud));
     }
 
     private boolean isOrphaned(Instance remote, Set<String> localInstances) {
-        String instanceName = remote.getName();
-        logger.log(Level.FINEST, "Checking instance " + instanceName);
-        return !localInstances.contains(instanceName);
+        if (localInstances.contains(remote.getName())) {
+            return false;
+        }
+        String nodeInUseTs = remote.getLabels().get(NODE_IN_USE_LABEL_KEY);
+        if (nodeInUseTs == null) {
+            return false;
+        }
+        long lastUsed = Long.parseLong(nodeInUseTs);
+        var dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSXXX");
+        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        boolean isOrphan = lastUsed < System.currentTimeMillis() - RECURRENCE_PERIOD * LOST_MULTIPLIER;
+        logger.log(
+                Level.FINEST,
+                "Instance " + remote.getName() + " last used at: " + dateFormat.format(lastUsed) + ", isOrphan: "
+                        + isOrphan);
```

Comment:
Think about using `hudson.Util.getTimeSpanString(long)`, put the formatting logic behind a check for `FINEST` level, either `logger.isLoggable(Level.FINEST)` or `java.util.logging.Logger#finest(java.util.function.Supplier<java.lang.String>)`

Reply:
You can also use `java.time`, which has nicer type-safe comparison operations, and `toString`s which are at least sort of human-readable.

Reply:
currently adopted the `OffsetDateTime` with using the `.isBefore` for comparison and human-readable format for label and same thing in log.

noted about not putting computation inside the log with level filter / supplier, thank you